src/Components/Components/src/NavigationManager.cs

```diff
@@ -103,7 +104,7 @@ protected set
     /// <param name="uri">The destination URI. This can be absolute, or relative to the base URI
     /// (as returned by <see cref="BaseUri"/>).</param>
     /// <param name="forceLoad">If true, bypasses client-side routing and forces the browser to load the new page from the server, whether or not the URI would normally be handled by the client-side router.</param>
-    public void NavigateTo(string uri, bool forceLoad) // This overload is for binary back-compat with < 6.0
+    public void NavigateTo([StringSyntax(StringSyntaxAttribute.Uri, UriKind.RelativeOrAbsolute)] string uri, bool forceLoad) // This overload is for binary back-compat with < 6.0
```

Comment:
None of the StringSyntax usages in dotnet/runtime - https://cs.github.com/dotnet/runtime?q=StringSyntaxAttribute.Uri - are passing UriKind itself as an argument. For example, https://github.com/dotnet/runtime/blob/960e4d723c27a5407dc691d06e546bc455a9c4a5/src/libraries/System.Private.Uri/src/System/Uri.cs#L408

Also, passing in `UriKind.RelativeOrAbsolute` doesn't seem useful as I think that would be the default.

@stephentoub @CyrusNajmabadi What should the argument be with `StringSyntaxAttribute.Uri`? Could whatever future tooling that uses that parameter support both a string that refers to another parameter or `UriKind` value?

Reply:
[`StringSyntaxAttribute.Regex` in dotnet/runtime](https://cs.github.com/dotnet/runtime?q=StringSyntaxAttribute.Regex) is happy with either which answers my question!